07.Rmd

```diff
@@ -1,37 +1,573 @@
+  gam.m3,
+  residuals = FALSE,
+  all.terms = TRUE,
+  shade = TRUE,
+  shade.col = 2,
+  rug = TRUE,
+  scale = 0
+)
+
+```
+
+A comparable model, built with lm and natural cubic splines, plotted as the GAM contributions to the `wage`:
+
+```{r chapter7-5-2, fig.asp=0.8}
+gam::plot.Gam(
+  lm(wage ~ ns(year, df = 4) + ns(age, df = 5) + education, data = Wage),
+  residuals = FALSE,
+  rugplot = TRUE,
+  se = TRUE,
+  scale = 0
+)
+```
+
+<h4>Why Use GAMs?</h4>
+
+There are at least three good reasons why you want to use GAM: interpretability, flexibility/automation, and regularization. Hence, when your model contains nonlinear effects, GAM provides a regularized and interpretable solution, while other methods generally lack at least one of these three features. In other words, GAMs strike a nice balance between the interpretable, yet biased, linear model, and the extremely flexible, “black box” learning algorithms.
+
+- Coefficients not that interesting; fitted functions are.
+
+- Can mix terms — some linear, some nonlinear
+
+- Can use smoothing splines AND local regression as well
+
+- GAMs are simply additive, although low-order interactions can be included in a natural way using, e.g. bivariate smoothers or interactions
+
+<h4>Fitting GAMs in R</h4>
+
+The two main packages in R that can be used to fit generalized additive models are `gam` and `mgcv`. The `gam` package was written by Trevor Hastie and is more or less frequentist. The `mgcv` package was written by Simon Wood, and, while it follows the same framework in many ways, it is much more general because it considers GAM to be any penalized GLM (and Bayesian). The differences are described in detail in the documentation for `mgcv`.
+
+I discovered that `gam` and `mgcv` do not work well when loaded at the same time. Restart the R session if you want to switch between the two packages – detaching one of the packages is not sufficient.
+
+An example of a classification GAM model:
+
+```{r chapter7-5-3, fig.asp=0.8}
+gam.lr <-
+  mgcv::gam(I(wage > 250) ~ as.numeric(year) + s(as.numeric(age), k = 5) + education, data = Wage)
+
+summary(gam.lr)
+
+par(mfrow = c(1, 3))
+mgcv::plot.gam(
+  gam.lr,
+  residuals = FALSE,
+  all.terms = TRUE,
+  shade = TRUE,
+  shade.col = 2,
+  rug = TRUE,
+  scale = 0
+)
+
+```
 
 ## Meeting Videos
 
 ### Cohort 1
 
-`r knitr::include_url("https://www.youtube.com/embed/EvpA_jXAe2g")`
+`r knitr::include_url("https://youtu.be/EvpA_jXAe2g")`
+
+<details>
+<summary> Meeting chat log </summary>
+
+```
+00:51:58    Jon Harmon (jonthegeek):    Because I'm obsessed now with "Why ξ?":
+"KS" for "knot/spline" = "ks" = "ξ"?
+I'm going to try to use that now, at least 🙃
+00:54:45    Laura Rose: 👍
+01:01:55    Federica Gazzelloni:    h(x,ξ) taylor : https://en.wikipedia.org/wiki/Taylors_theorem
+
+```
+</details>
+
+### Cohort 2
+
+`r knitr::include_url("https://youtu.be/wIj-rgxB-9A")`
 
 <details>
 <summary> Meeting chat log </summary>
 
 ```
+<<<<<<< HEAD
```

Comment:
There's some merge noise in this. Get rid of the HEAD line, and the === line, and the repeat of the cohort 1 stuff. Make sure it renders on your machine and looks right (ctrl-shift-k on PC, probably command-shift-k on Mac).